src/app/shared/directives/hide-if-not-authenticated.directive.spec.ts

```diff
@@ -0,0 +1,43 @@
+import { Component, Injectable, TemplateRef, ViewContainerRef } from '@angular/core';
+import { HideIfNotAuthenticatedDirective } from './hide-if-not-authenticated.directive';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { AuthenticationService } from '../services/authentication.service';
+
+describe('HideIfNotAuthenticatedDirective', () => {
+
+  @Injectable()
+  class MockService extends AuthenticationService {
+
+  }
```

Comment:
The services are mocked a little bit differently
You can use features of Jasmine directly 
Like this
`const mockService = jasmine.createSpyObj('AuthenticationService ', ['login', 'logout'...]);
`
And then 
`mockService.login.and.returnValue('mocked value');`

Reply:
OK, I didn't know how to mock a service, thanks